integration-tests/tests/bond_unbond.rs

```diff
@@ -14,8 +23,29 @@ fn workers_can_claim_stake() {
 
     let before_balances = test_utils::query_balances(&protocol.app, &workers);
 
+    let new_workers = test_utils::create_new_workers_vec(
+        chains.clone(),
+        vec![("worker3".to_string(), 3), ("worker4".to_string(), 4)],
+    );
+
+    test_utils::register_workers(&mut protocol, &new_workers, min_worker_bond);
+
     test_utils::deregister_workers(&mut protocol, &workers);
```

Comment:
I think this function just deauthorizes right? If so, can we have a test where the worker is not deauthorized, but simply deregisters chain support? So worker deregisters chain support, and then tries to unbond while still in the active workerset, which should fail.

Reply:
Added a test for it.